### PR TITLE
write tasks API using RTK query

### PR DESCRIPTION
**Pull request by bhtibrewal**
**Issue: #495**

### CODE REVIEW

File: src/app/services/tasksApi.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                    ),
+                };
+            },
+        }),
+        getMineTasks: builder.query<TasksQueryResponse, void>({
+            query: () => '/tasks/mine',
+            providesTags: ['Tasks'],
+        }),
+    }),
+});
```
Comment (pallabez) [inline src/app/services/tasksApi.ts:28]: Nice

File: src/pages/index.tsx [hunk 2]
```diff
@@ -73,8 +47,7 @@ async function updateCardContent(id: string, cardDetails: task) {
 
 const Index: FC = () => {
     const { state: appState } = useAppContext();
-    const [filteredTask, setFilteredTask] = useState<any>([]);
-    const { response, error, isLoading } = useFetch(TASKS_URL);
+    const { data: response, isError, isLoading } = useGetAllTasksQuery();
```
Comment (pallabez) [inline src/pages/index.tsx:50]:
```suggestion
    const { data: tasks, isError, isLoading } = useGetAllTasksQuery();
```

File: src/pages/index.tsx [hunk 3]
```diff
@@ -83,30 +56,17 @@ const Index: FC = () => {
...
+    const tasksGroupedByStatus = updateTasksStatus(
+        response?.tasks || []
+    ).reduce((acc: Record<string, task[]>, curr: task) => {
+        return acc[curr.status as keyof task]
+            ? {
+                  ...acc,
+                  [curr.status]: [...acc[curr.status as keyof task], curr],
+              }
+            : { ...acc, [curr.status]: [curr] };
+    }, {});
```
Comment (pallabez) [inline src/pages/index.tsx:70]: Thanks for the refactor

Comment (bhtibrewal) [reply src/pages/index.tsx:70]: Will address that in further iterations, Bharti has also wrote some util function for this, I will check it with that in the next PR

File: src/pages/index.tsx (hunk 2 above)
Comment (bhtibrewal) [reply src/pages/index.tsx:50]: type TasksQueryResponse = {
    message: string;
    tasks: task[];
};
it also has message, that's why I am using response

Comment (pallabez) [reply src/pages/index.tsx:50]: We don't message right? We can remove that in transformResponse.

File: src/pages/index.tsx (hunk 3 above)
Comment (pallabez) [reply src/pages/index.tsx:70]: Got it.

File: src/pages/index.tsx (hunk 2 above)
Comment (bhtibrewal) [reply src/pages/index.tsx]: done

File: src/app/services/tasksApi.ts (hunk 1 above)
Comment (bhtibrewal) [reply src/app/services/tasksApi.ts]: thanks

File: src/pages/index.tsx [hunk 4]
```diff
@@ -136,18 +96,18 @@ const Index: FC = () => {
             <Head title="Tasks" />
             <TasksProvider>
                 <div className={classNames.container}>
-                    {!!error && <p>{TASKS_FETCH_ERROR_MESSAGE}</p>}
+                    {isError && <p>{TASKS_FETCH_ERROR_MESSAGE}</p>}
```
Comment (vvaibhavdesai) [inline src/pages/index.tsx:99]: the optional chaining looks too much nested please fix this in a different PR

Comment (bhtibrewal) [reply src/pages/index.tsx:99]: okay

**Review by pallabez — APPROVED** [review]
LGTM

**Review by vvaibhavdesai — APPROVED** [review]

**Pull request merged**